File: src/backend/dotnet/TodoShared/Models/TodoItemAttachment.cs
namespace TodoShared.Models;

public class TodoItemAttachment
{
    public int Id { get; set; }
    public int TodoItemId { get; set; }
    public int FileId { get; set; }
    public DateTime CreatedAt { get; set; }
    public int? CreatedByUserId { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public int? UpdatedByUserId { get; set; }

    public TodoItem? TodoItem { get; set; }
    public FileEntity? File { get; set; }
}

File: src/backend/go/internal/models/todo_item_attachment.go
package models

import "time"

// TodoItemAttachment links an existing uploaded file to a todo item.
type TodoItemAttachment struct {
	ID              uint       `gorm:"primarykey;autoIncrement" json:"id"`
	TodoItemID      uint       `gorm:"not null;uniqueIndex:uq_todo_item_attachments_todo_file" json:"todoItemId"`
	FileID          uint       `gorm:"not null;uniqueIndex:uq_todo_item_attachments_todo_file" json:"fileId"`
	CreatedAt       time.Time  `gorm:"autoCreateTime" json:"createdAt"`
	CreatedByUserID *uint      `gorm:"index" json:"createdByUserId"`
	UpdatedAt       *time.Time `gorm:"autoUpdateTime" json:"updatedAt"`
	UpdatedByUserID *uint      `gorm:"index" json:"updatedByUserId"`
	TodoItem        TodoItem   `gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE;foreignKey:TodoItemID" json:"-"`
	File            File       `gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE;foreignKey:FileID" json:"-"`
	CreatedByUser   *User      `gorm:"constraint:OnUpdate:CASCADE,OnDelete:SET NULL;foreignKey:CreatedByUserID" json:"-"`
	UpdatedByUser   *User      `gorm:"constraint:OnUpdate:CASCADE,OnDelete:SET NULL;foreignKey:UpdatedByUserID" json:"-"`
}

func (TodoItemAttachment) TableName() string { return "todo_item_attachments" }
